pkg/component/observability/opentelemetry/opentelemetryoperator/opentelemetryoperator.go

```diff
@@ -0,0 +1,407 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package opentelemetryoperator
+
+import (
+	"context"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	vpaautoscalingv1 "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/component"
+	"github.com/gardener/gardener/pkg/resourcemanager/controller/garbagecollector/references"
+	"github.com/gardener/gardener/pkg/utils"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+)
+
+const (
+	// OperatorManagedResourceName is the name of the OpenTelemetry Operator managed resource.
+	OperatorManagedResourceName = "opentelemetry-operator"
+	name                        = "opentelemetry-operator"
+	roleName                    = "gardener.cloud:opentelemetry:opentelemetry-operator"
+)
+
+// Values keeps values for the OpenTelemetry Operator.
+type Values struct {
+	// Image is the image of the OpenTelemetry Operator.
+	Image string
+	// PriorityClassName is the name of the priority class of the OpenTelemetry Operator.
+	PriorityClassName string
+}
+
+type opentelemetryOperator struct {
```

Comment:
```suggestion
type openTelemetryOperator struct {
```

or maybe just `operator`, to make it shorter?